lib/algos/list/delimiterMaster.js

```diff
@@ -95,12 +95,29 @@ class DelimiterMaster extends Delimiter {
              *   NextMarker to the common prefix instead of the whole key
              *   value. (TODO: remove this test once ZENKO-1048 is fixed)
              *   */
-            if (key === this.prvKey || key === this[this.nextContinueMarker] ||
-                (this.delimiter &&
-                key.startsWith(this[this.nextContinueMarker]))) {
+            if (key === this.prvKey || key === this[this.nextContinueMarker]) {
                 /* master version already filtered */
                 return FILTER_SKIP;
             }
+
+            if (this.delimiter && key.startsWith(this[this.nextContinueMarker])) {
+                // If this[continueMarker] ends with the delimiter then we have to treat it differently.
+                const candidate = key.split(this.delimiter);
+                const len = candidate.length;
+                if (len === 1) {
+                    // case key=foo: this is a version of a previously seen master key.
```

Comment:
I don't understand the logic, can you try to clarify? Basically I don't see the relationship between not having any delimiter character in the key (hence `len === 1` when we split) and this key being a version of a previously seen master key. I see that a similar case is skipped below when the key ends with a delimiter, but I think if the logic is correct it could deserve some clarification.